lib/elixir/lib/range.ex

```diff
@@ -56,6 +56,41 @@ defmodule Range do
             "got: #{inspect(first)}..#{inspect(last)}"
   end
 
+  @doc """
+  Checks if two ranges overlap.
+
+  ## Examples
+
+      iex> Range.overlap?(1..7, 5..9)
+      true
+      iex> Range.overlap?(7..1, 5..9)
+      true
+      iex> Range.overlap?(1..7, 9..5)
+      true
+      iex> Range.overlap?(7..1, 9..5)
+      true
+      iex> Range.overlap?(0..1, 1..2)
+      true
+      iex> Range.overlap?(1..1, 1..1)
+      true
+      iex> Range.overlap?(1..5, 6..9)
+      false
+      iex> Range.overlap?(9..7, 3..1)
+      false
+  """
+  @spec overlap?(t, t) :: boolean
+  def overlap?(first1..last1, first2..last2) when first1 <= last1 and first2 <= last2 do
+    !(last2 < first1 || last1 < first2)
```

Comment:
```suggestion
    not(last2 < first1 or last1 < first2)
```

Since they are all booleans, we should prefer to use the strict boolean operators. :)